components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -394,17 +394,27 @@ public void updateOauthSecretKey(String consumerKey) throws IdentityOAuthAdminEx
 
     private void updateAppAndRevokeTokensAndAuthzCodes(String consumerKey, Properties properties) throws IdentityOAuthAdminException {
         TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
-
+        int countToken = 0;
         try {
-            Set<String> accessTokens = tokenMgtDAO.getActiveTokensForConsumerKey(consumerKey);
+            Set<AccessTokenDO> activeDetailedTokens = tokenMgtDAO.getActiveDetailedTokensForConsumerKey(consumerKey);
+            String[] accessTokens=new String[activeDetailedTokens.size()];
+            String cacheKeyString;
             if (OAuthServerConfiguration.getInstance().isCacheEnabled()) {
                 OAuthCache oauthCache = OAuthCache.getInstance();
-                for (String accessToken : accessTokens) {
-                    OAuthCacheKey cacheKey = new OAuthCacheKey(accessToken);
-                    oauthCache.clearCacheEntry(cacheKey);
+                for (AccessTokenDO detailToken: activeDetailedTokens) {
+                    String token = detailToken.getAccessToken();
+                    accessTokens[countToken]=token;
+                    countToken ++;
+                    for(String singleScope: detailToken.getScope()) {
+                        cacheKeyString = consumerKey + ":" + detailToken.getAuthzUser().toString() + ":" + singleScope;
```

Comment:
We are creating the cache key with the whole scope string when adding to the cache. Refer https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java#L95-L104